packages/@aws-cdk/aws-appmesh/lib/virtual-gateway-listener.ts

```diff
@@ -0,0 +1,227 @@
+import * as cdk from '@aws-cdk/core';
+import { CfnVirtualGateway } from './appmesh.generated';
+import { validateHealthChecks } from './private/utils';
+import { HealthCheck, Protocol } from './shared-interfaces';
+
+/**
+ * Represents the properties needed to define HTTP Listeners for a VirtualGateway
+ */
+export interface HttpGatewayListenerProps {
+  /**
+   * Port to listen for connections on
+   *
+   * @default - 8080
+   */
+  readonly port?: number
+
+  /**
+   * The health check information for the listener
+   *
+   * @default - no healthcheck
+   */
+  readonly healthCheck?: HealthCheck;
+}
+
+/**
+ * Represents the properties needed to define GRPC Listeners for a VirtualGateway
+ */
+export interface GrpcGatewayListenerProps {
+  /**
+   * Port to listen for connections on
+   *
+   * @default - 8080
+   */
+  readonly port?: number
+
+  /**
+   * The health check information for the listener
+   *
+   * @default - no healthcheck
+   */
+  readonly healthCheck?: HealthCheck;
+}
+
+/**
+ * Properties for a VirtualGateway listener
+ */
+export interface VirtualGatewayListenerConfig {
+  /**
+   * Single listener config for a VirtualGateway
+   */
+  readonly listener: CfnVirtualGateway.VirtualGatewayListenerProperty,
+}
+
+/**
+ * Represents the properties needed to define listeners for a VirtualGateway
+ */
+export abstract class VirtualGatewayListener {
+  /**
+   * Returns an HTTP Listener for a VirtualGateway
+   */
+  public static httpGatewayListener(props: HttpGatewayListenerProps = {}): VirtualGatewayListener {
+    return new HttpGatewayListener(props);
+  }
+
+  /**
+   * Returns an HTTP2 Listener for a VirtualGateway
+   */
+  public static http2GatewayListener(props: HttpGatewayListenerProps = {}): VirtualGatewayListener {
+    return new Http2GatewayListener(props);
+  }
+
+  /**
+   * Returns a GRPC Listener for a VirtualGateway
+   */
+  public static grpcGatewayListener(props: GrpcGatewayListenerProps = {}): VirtualGatewayListener {
+    return new GrpcGatewayListener(props);
+  }
+
+  /**
+   * Protocol the listener implements
+   */
+  protected abstract protocol: Protocol;
+
+  /**
+   * Port to listen for connections on
+   */
+  protected abstract port: number;
+
+  /**
+   * Health checking strategy upstream nodes should use when communicating with the listener
+   */
+  protected abstract healthCheck?: HealthCheck;
+
+  /**
+   * Called when the GatewayListener type is initialized. Can be used to enforce
+   * mutual exclusivity
+   */
+  public abstract bind(scope: cdk.Construct): VirtualGatewayListenerConfig;
+
+  protected renderHealthCheck(hc: HealthCheck): CfnVirtualGateway.VirtualGatewayHealthCheckPolicyProperty | undefined {
+    if (hc.protocol === Protocol.TCP) {
```

Comment:
Is this `if` missing ` && hc.path`?

Reply:
TCP HC are not permitted for virtual gateways

Reply:
I've updated the Error Message